File: dotnet-desktop-guide/net/winforms/controls-design/snippets/how-to-designer-properties-shouldserialize-reset/vb/CompassRose.vb
'<control>
Public Class CompassRose

    '<property>
    Public Property Direction As Directions = Directions.None
    '</property>

    '<reset>
    Private Sub ResetDirection()
        Direction = Directions.None
    End Sub
    '</reset>

    '<shouldserialize>
    Private Function ShouldSerializeDirection() As Boolean
        Return Direction <> Directions.None
    End Function
    '</shouldserialize>

End Class
'</control>

File: dotnet-desktop-guide/net/winforms/controls-design/snippets/how-to-designer-properties-shouldserialize-reset/csharp/CompassRose.cs
namespace UserControlProject;

// <control>
public partial class CompassRose : UserControl
{
    // <property>
    public Directions Direction { get; set; } = Directions.None;
    // </property>

    public CompassRose() =>
        InitializeComponent();

    // <reset>
    private void ResetDirection() =>
        Direction = Directions.None;
    // </reset>

    // <shouldserialize>
    private bool ShouldSerializeDirection() =>
        Direction != Directions.None;
    // </shouldserialize>
}
// </control>
